README.md

```diff
@@ -28,7 +28,256 @@ UpSet: Visualization of Intersecting Sets
+- `provVis` (optional): [Sidebar options](#sidebar-options) for the provenance visualization sidebar. See [Trrack-Vis](https://github.com/Trrack/trrackvis) for more information about Trrack provenance visualization.
+- `elementSidebar` (optional): [Sidebar options](#sidebar-options) for the element visualization sidebar. This sidebar is used for element queries, element selection datatable, and supplimental plot generation.
+- `altTextSidebar` (optional): [Sidebar options](#sidebar-options) for the text description sidebar. This sidebar is used to display the generated text descriptions for an Upset 2.0 plot, given that the `generatedAltText` function is provided.
+- `generateAltText` (optional)(`() => Promise<AltText>`): Async function which should return a generated AltText object. See [Alt Text Generation](#alt-text-generation) for more information about Alt Text generation.
+
+##### Configuration options
+
+If no configuration options are provided, the default will be:
+
+- No aggregation
+- Sort rows by `Size - Descending`
+- Sort visible sets by `Alphabetical`
+- Minimum degree filter: 0
+- Maximum degree filter: 6
+- Show the "no set" row
+- Hide empty rows
+
+Example of full configuration (grammar) JSON produced for default Simpsons dataset:
+
+```json
+{
+  "plotInformation": {
+    "description": "",
+    "sets": "",
+    "items": ""
+  },
+  "horizontal": false,
+  "firstAggregateBy": "None",
+  "firstOverlapDegree": 2,
+  "secondAggregateBy": "None",
+  "secondOverlapDegree": 2,
+  "sortVisibleBy": "Alphabetical",
+  "sortBy": "Size",
+  "sortByOrder": "Descending",
+  "filters": {
+    "maxVisible": 6,
+    "minVisible": 0,
+    "hideEmpty": true,
+    "hideNoSet": false
+  },
+  "visibleSets": [
+    "Set_School",
+    "Set_Blue Hair",
+    "Set_Duff Fan",
+    "Set_Evil",
+    "Set_Male",
+    "Set_Power Plant"
+  ],
+  "visibleAttributes": [
+    "Age"
+  ],
+  "bookmarkedIntersections": [],
+  "collapsed": [],
+  "plots": {
+    "scatterplots": [],
+    "histograms": []
+  },
+  "allSets": [
+    {
+      "name": "Set_School",
+      "size": 6
+    },
+    {
+      "name": "Set_Blue Hair",
+      "size": 3
+    },
+    {
+      "name": "Set_Duff Fan",
+      "size": 6
+    },
+    {
+      "name": "Set_Evil",
+      "size": 6
+    },
+    {
+      "name": "Set_Male",
+      "size": 18
+    },
+    {
+      "name": "Set_Power Plant",
+      "size": 5
+    }
+  ],
+  "selected": null
+}
+```
+
+##### Sidebar options
+
+SidebarProps type:
+
+```js
+interface SidebarProps {
+  open: boolean;
+  close: () => void;
+}
+```
+
+##### Alt Text Generation
+
+Alt Text generation requires the use of a custom or imported Alt-Text generation function. In [upset.multinet.app](https://upset.multinet.app), we are using the Multinet API, which exposes an api call to the [upset-alttxt](https://pypi.org/project/upset-alttxt/) python package. This Python package 
```

Comment:
The difficulty with this is that the Multinet API is only accessible in the `app` package. We would be adding a dependency on multinet-api within the Upset react component

Reply:
Can we change that? It seems like that's within scope for `upset` since it provides the alttext pane. It's also pretty major in terms of not requiring users to figure out & write boilerplate to call the API and provide `generateAltText`. I'm guessing absent that change we just won't see many users providing alttexts for their upset implementations.

Reply:
And by "ADHD" moment, I was referring to the `This Python Package` at the end, which I assume you meant to turn into a full sentence? ;)